ui/views/tx/add/add-tx-view.js

```diff
@@ -30,38 +30,54 @@ function TxPropsBlock({title, description, optional = false, children}) {
 
 export default function AddTxView() {
     const [data, setData] = useState({
-            xdr: '',
-            network: 'public',
-            submit: false,
-            callback: '',
-            expires: '',
-            desiredSigners: []
-        }),
-        [error, setError] = useState(''),
-        [inProgress, setInProgress] = useState(false)
-
-    function setParam(param, value) {
+        xdr: '',
+        network: 'public',
+        submit: false,
+        callback: '',
+        expires: '',
+        desiredSigners: []
+    })
+    const [inProgress, setInProgress] = useState(false)
+
+    const setParam = useCallback((param, value) => {
         setData(prev => {
             const newData = {...prev, [param]: value}
             if (!isEqual(prev, newData)) return newData
             return prev
         })
-    }
+    }, [])
 
-    function storeTx() {
+    const storeTx = useCallback(() => {
         setInProgress(true)
         return submitTx(data)
             .then(res => {
-                console.log(res)
                 navigation.navigate(`/tx/${res.hash}`)
             })
             .catch(e => {
-                console.error(e)
-                setError(e.message)
+                setInProgress(false)
+                notify({type: 'error', message: e.message})
             })
-            .finally(() => setInProgress(false))
-    }
+    }, [data])
+
+    const changeNetwork = useCallback(n => {
+        setParam('network', n)
+    }, [setParam])
```

Comment:
```suggestion
    const changeNetwork = useCallback(n => setParam('network', n), [setParam])
```

Use inline invocations for such primitive code to avoid blowing up files unnecessarily.